### ISSUE TITLE
"Discount parameter did not have timestamp key" error when calling Purchases.purchaseDiscountedProduct() 

### ISSUE THREAD

**Issue by benmarsh** [RevenueCat/purchases-capacitor#168]
I have an issue when attempting to use Purchases.purchaseDiscountedProduct(). 

I resolve the product and discount from the relevant IDs, when when passing this through to purchaseDiscountedProduct I get the following error. From what I can see the timestamp is set and is available? Screenshot of the error is below.

Non-discounted purchases work correctly with the purchaseStoreProduct method. Brief example of the code I'm using is below too. I'm testing this locally with the app running on a real device iOS 17.1.2. Using SDK @revenuecat/purchases-capacitor v7.3.1 and on Capacitor v5.4.2.

![Screenshot 2023-12-18 at 15 26 16](https://github.com/RevenueCat/purchases-capacitor/assets/150741/4746537a-8934-44ba-ae69-c2c5be18297f)

```js
    async getProduct(identifier: string): Promise<PurchasesStoreProduct|null> {
        const products = await this.getProducts([identifier], PRODUCT_CATEGORY.NON_SUBSCRIPTION);
        if (products && products.length > 0) {
            return products[0];
        }
        return null;
    }

    private getDiscountForProduct(product: PurchasesStoreProduct, offerIdentifier: string) {
        const discount = product.discounts.find((discount) => {
            return discount.identifier === offerIdentifier;
        });
        return Purchases.getPromotionalOffer({
            product,
            discount
        });
    }

    async purchase(productIdentifier: string, offerIdentifier: string = null) {
        const product = await this.getProduct(productIdentifier);

        if (offerIdentifier) {
            const discount = await this.getDiscountForProduct(product, offerIdentifier);
            return Purchases.purchaseDiscountedProduct({
                product,
                discount
            });
        }

        return Purchases.purchaseStoreProduct({
            product
        });
    }
```

**Comment by benmarsh** [comment]
OK, went down a bit of a rabbit hole on this and managed to get it working.

In PurchasesPlugin.swift, and the purchaseDiscountedProduct method, the timestamp being passed in is numeric and the line... 
```swift 
guard let signedDiscounTimestamp = discount["timestamp"] as? String else {
```
...was returning signedDiscounTimestamp as `nil`. By changing the signedDiscounTimestamp cast to `as? Int` and then casting it back into a String now correctly passes the timestamp field to the CommonFunctionality.purchase method, as below;

```swift
@objc func purchaseDiscountedProduct(_ call: CAPPluginCall) {
        guard self.rejectIfPurchasesNotConfigured(call) else { return }
        guard let storeProduct = call.getOrRejectObject("product") else { return }
        guard let productId = storeProduct["identifier"] as? String else {
            call.reject("Product parameter did not have identifier key")
            return
        }
        guard let discount = call.getOrRejectObject("discount") else { return }
        guard let signedDiscounTimestamp = discount["timestamp"] as? Int else {
            call.reject("Discount parameter did not have timestamp key")
            return
        }
        let signedDiscounTimestampString: String? = String(signedDiscounTimestamp);
        CommonFunctionality.purchase(product: productId,
                                     signedDiscountTimestamp: signedDiscounTimestampString,
                                     completion: self.getCompletionBlockHandler(call))
    }
```

There's probably a cleaner way to so this in Swift but the above I had working on my device and correctly showing the native payment popup with the introductory offer displayed.

Can I request an update is made to the plugin to allow this functionality to work going forward? Thanks.

**Comment by benmarsh** [comment]
Have submitted a PR with a basic fix > https://github.com/RevenueCat/purchases-capacitor/pull/97

**Comment by tonidero** [comment]
Hey @benmarsh, thanks for reporting and then fixing this issue! Greatly appreciated! I took a look at your PR and I think we can get it merged. We will make a new release with the fix soon and add you to the list of contributors for the release.

Thanks again for the contribution!

**Comment by tonidero** [comment]
Hello again @benmarsh, we just released version 7.3.2 with your fix. Please take a look and let us know if you run into any issues. In the meantime, I'm closing this ticket. Thanks once again for your contribution!
